scripts/azureml-assets/azureml/assets/copy_assets.py

```diff
@@ -31,9 +34,33 @@ def copy_asset(asset_config: assets.AssetConfig,
         # Skip a released version
         return None
 
+    if check_previous_release:
+        # Skip if previous version was not released
+        previous_release_version = assets.get_latest_release_tag_version(asset_config, release_directory_root)
+
+        if previous_release_version is not None:
+            image = asset_config.extra_config_as_object().get_full_image_name()
+
+            # Check against MCR by making a manifest call to see if tag exists
+            (hostname, repo) = image.split("/", 1)
+
+            request = assets._get_manifest(previous_release_version, hostname, repo)
+
+            try:
+                response = assets._urlopen_with_retries(request)
+            except HTTPError as e:
+                if e.code == 404:
+                    logger.log_error(f"Image {image} does not exist in MCR")
+                else:
+                    logger.log_error(f"Unexpected error when looking for image {image} in MCR")
+                raise Exception(f"Failed to retrieve manifest for {repo}:{previous_release_version}: {e}")

```

Comment:
Indent

Reply:
Should we indent this Exception and then add an exit(1) after the logging error for a 404?

Reply:
Yes, I think that's a great approach.